src/lib.rs

```diff
@@ -29,6 +32,37 @@ pub struct Rect {
     pub height: u16
 }
 
+impl Rect {
+    /// Constructs new `Rect`.
+    pub fn new(left: u16, top: u16, width: u16, height: u16) -> Self {
+        Rect {
+            left: left,
+            top: top,
+            width: width,
+            height: height,
+        }
+    }
+
+    /// Constructs new zero-sized `Rect` placed at (0, 0).
+    pub fn new_empty() -> Self {
+        Rect {
+            left: 0,
+            top: 0,
+            width: 0,
+            height: 0,
+        }
+    }
+
+    /// Writes `Rect` to given stream.
```

Comment:
With `Message` trait? This is not a message. Without? That method would be unused generating warning.